ouroboros-network-api/src/Ouroboros/Network/BlockFetch/ConsensusInterface.hs

```diff
@@ -1,24 +1,35 @@
-{-# LANGUAGE DeriveFunctor #-}
-{-# LANGUAGE RankNTypes    #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveFunctor  #-}
+{-# LANGUAGE DeriveGeneric  #-}
+{-# LANGUAGE RankNTypes     #-}
 
 module Ouroboros.Network.BlockFetch.ConsensusInterface
   ( FetchMode (..)
+  , GenesisFetchMode (..)
   , BlockFetchConsensusInterface (..)
   , WhetherReceivingTentativeBlocks (..)
   , FromConsensus (..)
+  , ChainSelStarvation (..)
   ) where
 
 import Control.Monad.Class.MonadSTM
 import Control.Monad.Class.MonadTime (UTCTime)
+import Control.Monad.Class.MonadTime.SI (Time)
 
 import Data.Map.Strict (Map)
+import GHC.Generics (Generic)
 import GHC.Stack (HasCallStack)
+import NoThunks.Class (NoThunks)
 
 import Ouroboros.Network.AnchoredFragment (AnchoredFragment)
 import Ouroboros.Network.Block
 import Ouroboros.Network.SizeInBytes (SizeInBytes)
 
 
+-- | The fetch mode that the block fetch logic should use.
+data GenesisFetchMode = FetchModeGenesis | PraosFetchMode FetchMode
```

Comment:
It seems a bit confusing that `PraosFetchMode` is of type `GenesisFetchMode`. One idea would be to have rename stuff like this:
```haskell
data FetchMode = FetchModeGenesis | PraosFetchMode PraosFetchMode
data PraosFetchMode = FetchModeDeadline | FetchModeBulkSynk
```
But this would cause a larger diff of course.